status-im/status-react
# Synchronize your own messages in 1-to-1

Once two devices are paired, you should see your own messages in 1-to-1 chats on different devices.

Depends on: #6242 

If you have a device with a previous version it will see those messages as coming from you (a chat opened with yourself will receive those messages).

I don't think is a big deal as the user has control over their devices (they just have to upgrade the version), but let me know if you think otherwise.

Enabled only in nightlies & when dev-mode is on, same as pairing.

Only the last commit is relevant.

We allow now the user to set an arbitrary chat-id in the message, so this needs to be properly validated.
The logic for the validation is here https://github.com/status-im/status-react/blob/62db799d60fe5a852ad6c5285658092488ad7beb/src/status_im/chat/models/message.cljs#L171 and probably needs to be looked at carefully. 

### Step to tests

1) You need to have enabled development mode
2) Go to settings -> Devices
3) On android/ios click on `+`, desktop click on `pair` to advertise your device
4) Any other device should see the new device with a 3-words random name + device-type under `devices`
5) If you enable the device, it will send 1-to-1 & group messages to that device, make sure that devices are enabled both ways
6) New group chats will also be synchronized

The UI is still rough and pending designs.

It will only sync max 4 extra devices for now, we will work on the UI to make sure the user don't enable more.

Enables in nightlis in development mode

status: ready
